compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -693,8 +693,13 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
    */
   def eraseInfo(tp: Type, sym: Symbol)(using Context): Type =
     val tp1 = tp match
-      case tp: MethodicType => integrateContextResults(tp, contextResultCount(sym))
-      case _ => tp
+      case tp: MethodicType =>
+        integrateContextResults(tp, contextResultCount(sym))
+      case tp: ClassInfo
+      if defn.isTupleClass(sym) && tp.declaredParents.head.typeSymbol == defn.PairClass =>
+        tp.derivedClassInfo(declaredParents = defn.ObjectType :: tp.declaredParents.tail)
```

Comment:
Could this be handled by adding an entry to the specialErasure map instead? (https://github.com/lampepfl/dotty/blob/eb8773ecbedd32b7f7f6822696f3c3df3cde6930/compiler/src/dotty/tools/dotc/core/Definitions.scala#L1772-L1779)